src/main/java/emissary/core/BaseDataObject.java

```diff
@@ -440,10 +441,17 @@ public int enqueueCurrentForm(final String newForm) {
             throw new IllegalArgumentException("caller attempted to enqueue a null form value");
         }
 
+        checkForDuplicates(newForm);
```

Comment:
Thinking out loud here, is it better to remove the form or not add it? Essentially, you could be kicking a form to the back of the line - I don't know if that is good or bad. Possibly unintended consequences? Something to think about.

Reply:
The intention was to push the form to the top if for whatever reason it was encountered again, but I may have gone about it incorrectly.